typescript/sdk/src/token/EvmERC20WarpModule.ts

```diff
@@ -52,15 +60,105 @@ export class EvmERC20WarpModule extends HyperlaneModule<
+    );
+
+    return updateTransactions;
+  }
+
+  /**
+   * Updates an existing Warp route ISM with a given config.
+   *
+   * @param actualConfig - The on-chain router configuration, including the ISM configuration, and address.
+   * @param expectedConfig - The expected token router configuration, including the ISM configuration.
+   * @returns Ethereum transaction that need to be executed to update the ISM configuration.
+   */
+  async updateIsm(
+    actualConfig: TokenRouterConfig,
+    expectedConfig: TokenRouterConfig,
   ): Promise<AnnotatedEV5Transaction[]> {
-    throw Error('Not implemented');
+    const updateTransactions: AnnotatedEV5Transaction[] = [];
+    const actualDeployedIsm = (
+      actualConfig.interchainSecurityModule as DerivedIsmConfig
+    ).address;
+
+    // Try to update (may also deploy) Ism with the expected config
+    const {
+      deployedIsm: expectedDeployedIsm,
+      updateTransactions: ismUpdateTransactions,
+    } = await this.deployOrUpdateIsm(actualConfig, expectedConfig);
+
+    // If an ISM is updated in-place, push the update txs
+    updateTransactions.push(...ismUpdateTransactions);
+
+    // If a new ISM is deployed, push the setInterchainSecurityModule tx
+    if (actualDeployedIsm !== expectedDeployedIsm) {
+      const contractToUpdate = MailboxClient__factory.connect(
+        this.args.addresses.deployedTokenRoute,
+        this.multiProvider.getProvider(this.args.chain),
+      );
+      updateTransactions.push({
+        annotation: `Setting ISM for Warp Route to ${expectedDeployedIsm}`,
+        chainId: Number(this.multiProvider.getChainId(this.args.chain)),
+        to: contractToUpdate.address,
+        data: contractToUpdate.interface.encodeFunctionData(
+          'setInterchainSecurityModule',
+          [expectedDeployedIsm],
+        ),
+      });
+    }
+
+    return updateTransactions;
+  }
+
+  /**
+   * Updates or deploys the ISM using the provided configuration.
+   *
+   * @returns Object with deployedIsm address, and update Transactions
+   */
+  public async deployOrUpdateIsm(
+    actualConfig: TokenRouterConfig,
+    expectedConfig: TokenRouterConfig,
+  ): Promise<{
+    deployedIsm: Address;
+    updateTransactions: AnnotatedEV5Transaction[];
+  }> {
+    assert(
+      expectedConfig.interchainSecurityModule,
+      'Ism not derived correctly',
+    );
+    assert(
+      expectedConfig.ismFactoryAddresses,
+      'Ism Factories addresses not provided',
+    );
+
+    const ismModule = new EvmIsmModule(this.multiProvider, {
+      chain: this.args.chain,
+      config: expectedConfig.interchainSecurityModule,
+      addresses: {
+        ...expectedConfig.ismFactoryAddresses,
+        mailbox: expectedConfig.mailbox,
+        deployedIsm: (actualConfig.interchainSecurityModule as DerivedIsmConfig)
+          .address,
+      },
+    });
+    const updateTransactions = await ismModule.update(
+      expectedConfig.interchainSecurityModule,
+    );
```

Comment:
ahh nice I see what you did here! cool way of reusing the update logic without having the ism module embedded as a property of the class